x/registry/types/who_is.go

```diff
@@ -2,214 +2,93 @@ package types
 
 import (
 	"fmt"
+	"strings"
+	"time"
 
-	"github.com/duo-labs/webauthn/protocol"
-	"github.com/duo-labs/webauthn/webauthn"
 	"github.com/sonr-io/sonr/pkg/did"
-	rt "go.buf.build/grpc/go/sonr-io/blockchain/registry"
 )
 
-func NewWhoIsFromBuf(doc *rt.WhoIs) *WhoIs {
-	return &WhoIs{
-		Type:        WhoIs_Type(doc.GetType()),
-		Name:        doc.GetName(),
-		Did:         doc.GetDid(),
-		Document:    doc.GetDocument(),
-		Metadata:    doc.GetMetadata(),
-		Credentials: NewCredentialListFromBuf(doc.GetCredentials()),
+// ContainsAlias checks if the alias is in the list of aliases of the whois
+func (w *WhoIs) ContainsAlias(target string) bool {
+	for _, a := range w.Alias {
+		if a.GetName() == target {
+			return true
+		}
 	}
+	return false
 }
 
-func NewWhoIsToBuf(doc *WhoIs) *rt.WhoIs {
-	return &rt.WhoIs{
-		Type:        rt.WhoIs_Type(doc.GetType()),
-		Name:        doc.GetName(),
-		Did:         doc.GetDid(),
-		Document:    doc.GetDocument(),
-		Metadata:    doc.GetMetadata(),
-		Credentials: NewCredentialListToBuf(doc.GetCredentials()),
+// ContainsController checks if the controller is in the list of controllers of the whois
+func (w *WhoIs) ContainsController(target string) bool {
+	// Validates DID String
+	if _, err := did.ParseDID(target); err != nil {
```

Comment:
This check is extraneous because a malformed controller won't exist in the list anyway.